File: ngDesk/ngDesk-Rest/src/main/java/com/ngdesk/companies/EmailList.java
package com.ngdesk.companies;

import java.sql.Timestamp;
import java.util.List;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Pattern;
import javax.validation.constraints.Size;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.ngdesk.modules.rules.Condition;

public class EmailList {
	
	@JsonProperty("EMAIL_LIST_ID")
	private String emailListId;

	@JsonProperty("NAME")
	@NotNull(message = "EMAIL_LIST_NAME_NOT_NULL")
	private String name;
	
	@JsonProperty("DESCRIPTION")
	private String description;
	
	@JsonProperty("CONDITIONS")
	@Valid
	private List<Condition> conditions;

	@JsonProperty("DATE_CREATED")
	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
	private Timestamp dateCreated;
	
	@JsonProperty("DATE_UPDATED")
	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
	private Timestamp dateUpdated;

	@JsonProperty("LAST_UPDATED_BY")
	private String lastUpdatedBy;

	@JsonProperty("CREATED_BY")
	private String createdBy;
	
	public EmailList() {
		
	}

	public EmailList(String emailListId,
			@NotNull(message = "EMAIL_LIST_NAME_NOT_NULL") String name,
			String description, @Valid List<Condition> conditions, Timestamp dateCreated,
			Timestamp dateUpdated, String lastUpdatedBy, String createdBy) {
		super();
		this.emailListId = emailListId;
		this.name = name;
		this.description = description;
		this.conditions = conditions;
		this.dateCreated = dateCreated;
		this.dateUpdated = dateUpdated;
		this.lastUpdatedBy = lastUpdatedBy;
		this.createdBy = createdBy;
	}

	public String getEmailListId() {
		return emailListId;
	}

	public void setEmailListId(String emailListId) {
		this.emailListId = emailListId;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public List<Condition> getConditions() {
		return conditions;
	}

	public void setConditions(List<Condition> conditions) {
		this.conditions = conditions;
	}

	public Timestamp getDateCreated() {
		return dateCreated;
	}

	public void setDateCreated(Timestamp dateCreated) {
		this.dateCreated = dateCreated;
	}

	public Timestamp getDateUpdated() {
		return dateUpdated;
	}

	public void setDateUpdated(Timestamp dateUpdated) {
		this.dateUpdated = dateUpdated;
	}

	public String getLastUpdatedBy() {
		return lastUpdatedBy;
	}

	public void setLastUpdatedBy(String lastUpdatedBy) {
		this.lastUpdatedBy = lastUpdatedBy;
	}

	public String getCreatedBy() {
		return createdBy;
	}

	public void setCreatedBy(String createdBy) {
		this.createdBy = createdBy;
	}

}

File: ngDesk-UI/src/app/models/email-list.ts
import { Condition } from './condition';

export class EmailList {
	constructor(
		private NAME: string,
		private DESCRIPTION: string,
		private CONDITIONS: Condition[],
		private EMAIL_LIST_ID?: string,
		private DATE_CREATED?: string,
		private DATE_UPDATED?: string,
		private LAST_UPDATED_BY?: string,
		private CREATED_BY?: string
	) {}

	public get name() {
		return this.NAME;
	}

	public set name(name: string) {
		this.NAME = name;
	}

	public get description() {
		return this.DESCRIPTION;
	}

	public set description(description: string) {
		this.DESCRIPTION = description;
	}

	public get emailListId() {
		return this.EMAIL_LIST_ID;
	}

	public set emailListId(emailListId: string) {
		this.EMAIL_LIST_ID = emailListId;
	}

	public get conditions() {
		return this.CONDITIONS;
	}

	public set conditions(conditions: Condition[]) {
		this.CONDITIONS = conditions;
	}

	public get dateCreated() {
		return this.DATE_CREATED;
	}

	public set dateCreated(dateCreated: string) {
		this.DATE_CREATED = dateCreated;
	}

	public get dateUpdated() {
		return this.DATE_UPDATED;
	}

	public set dateUpdated(dateUpdated: string) {
		this.DATE_UPDATED = dateUpdated;
	}

	public get lastUpdatedBy() {
		return this.LAST_UPDATED_BY;
	}

	public set lastUpdatedBy(lastUpdatedBy: string) {
		this.LAST_UPDATED_BY = lastUpdatedBy;
	}

	public get createdBy() {
		return this.CREATED_BY;
	}

	public set createdBy(createdBy: string) {
		this.CREATED_BY = createdBy;
	}
}
